numba/core/interpreter.py

```diff
@@ -2994,21 +3010,43 @@ def op_MAKE_CLOSURE(self, inst, name, code, closure, annotations,
         self.op_MAKE_FUNCTION(inst, name, code, closure, annotations,
                               kwdefaults, defaults, res)
 
-    def op_LOAD_CLOSURE(self, inst, res):
-        n_cellvars = len(self.code_cellvars)
-        if inst.arg < n_cellvars:
-            name = self.code_cellvars[inst.arg]
-            try:
-                gl = self.get(name)
-            except NotDefinedError:
-                msg = "Unsupported use of op_LOAD_CLOSURE encountered"
-                raise NotImplementedError(msg)
-        else:
-            idx = inst.arg - n_cellvars
-            name = self.code_freevars[idx]
-            value = self.get_closure_value(idx)
-            gl = ir.FreeVar(idx, name, value, loc=self.loc)
-        self.store(gl, res)
+    if PYVERSION == (3, 11):
+
+        def op_LOAD_CLOSURE(self, inst, res):
+            n_cellvars = len(self.code_cellvars)
+            arg = inst.arg - len(self.code_locals)
+            if arg < n_cellvars:
+                name = self.code_cellvars[arg]
+                try:
+                    gl = self.get(name)
+                except NotDefinedError:
+                    msg = "Unsupported use of op_LOAD_CLOSURE encountered"
+                    raise NotImplementedError(msg)
+            else:
+                idx = arg - n_cellvars
+                name = self.code_freevars[idx]
+                value = self.get_closure_value(idx)
+                gl = ir.FreeVar(idx, name, value, loc=self.loc)
+            self.store(gl, res)
+
+    elif PYVERSION < (3, 11):
+        def op_LOAD_CLOSURE(self, inst, res):
+            n_cellvars = len(self.code_cellvars)
+            if inst.arg < n_cellvars:
+                name = self.code_cellvars[inst.arg]
+                try:
+                    gl = self.get(name)
+                except NotDefinedError:
+                    msg = "Unsupported use of op_LOAD_CLOSURE encountered"
+                    raise NotImplementedError(msg)
+            else:
+                idx = inst.arg - n_cellvars
+                name = self.code_freevars[idx]
+                value = self.get_closure_value(idx)
+                gl = ir.FreeVar(idx, name, value, loc=self.loc)
+            self.store(gl, res)
+    else:
+        raise NotImplementedError(PYVERSION)
```

Comment:
I don't think this is quite right, but it is fixed in 485a1e89 on #8621.

Reply:
stay as is in this pr